src/app/config/projectsJsonData.ts

```diff
@@ -49,15 +49,11 @@ export default class projectsJsonData {
     return this.m_projectJsonData.projectTypes[_.toLower(projectType)].templates.javascript != undefined && this.m_projectJsonData.projectTypes[_.toLower(projectType)].templates.typescript != undefined;
   }
 
-  getHostTemplateNames(isSsoProject: boolean) {
+  getHostTemplateNames(projectType: string) {
     let hosts: string[] = [];
-    for (let key in this.m_projectJsonData.hostTypes) {
-      if (isSsoProject) {
-        if (key === 'excel' || key === 'word' || key === 'powerpoint') {
-          hosts.push(this.m_projectJsonData.hostTypes[key].displayname);
-        }
-      } else {
-        hosts.push(this.m_projectJsonData.hostTypes[key].displayname);
+    for (let key in this.m_projectJsonData.projectTypes) {
```

Comment:
This isn't a change that you need to make but just saw it and wanted to comment.

TypeScript added an equivalent of the null-coalescing operator in C# (?.) which returns undefined if null or undefined. So it might be possible to change implementations such as this loop to be a more direct call:

```
const hosts: string[] = this.m_projectJsonData.projectTypes[key]?.projectTypes ?? [];
```

So if the mapping for the key returned undefined, it falls back to an empty array, and if defined, returns the array. This would avoid having to actually loop through, so O(1) instead of O(n).

```

Reply:
Not sure what's causing the failure on the Windows CI run. In fact the CI loop bails out immediately without testing anything

Reply:
CI loop passing now - must've been some intermittent issue with the CI agent machine